### PR TITLE
Add public API baselines for SiteExtensions, Testing, WebEncoders

### PR DESCRIPTION
**Pull request by Pilchie**

### CODE REVIEW

**Comment by Pilchie** [comment]
Part of #24347

File: src/Testing/src/AssemblyTestLog.cs [hunk 1]
```diff
@@ -5,6 +5,7 @@
 using System;
 using System.Collections.Generic;
 using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
```
Comment (dougbu) [inline src/Testing/src/AssemblyTestLog.cs:8]: This is a fairly big change given this isn't a shipping package. I'm not even sure why it's packable. Might want to revert all changes under src/Testing. Up to you.

I should have realized this when I wrote up #24347.

**Review by dougbu — APPROVED** [review]

**Pull request merged**